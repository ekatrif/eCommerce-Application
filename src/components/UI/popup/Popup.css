.popup {
  position: fixed;
  left: calc(50% - 100px);
  bottom: 125px;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 200px;
  height: 50px;
  color: #fff;
  background-color: #1F2827;
  box-shadow: 0px 0px 20px #1F2827;
  transition: all .3s ease;
  scale: 0;
}

.popup-active {
  scale: 1;
}