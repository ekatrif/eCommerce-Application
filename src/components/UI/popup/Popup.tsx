import React from 'react';
import { Alert, AlertTitle } from '@mui/material';
import Snackbar from '@mui/material/Snackbar';
import './Popup.css';

interface IPopupProps {
  active: boolean;
  setActive: React.Dispatch<React.SetStateAction<boolean>>;
  popupType: 'error' | 'warning' | 'info' | 'success';
  message: string;
}

const Popup: React.FC<IPopupProps> = ({ active, setActive, popupType, message }) => {
  const handleClose = (): void => {
    setActive(false);
  };

  return (
    <div>
      <Snackbar
        open={active}
        autoHideDuration={1000}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        onClose={handleClose}
      >
        <Alert severity={popupType} onClose={handleClose}>
          <AlertTitle>{popupType.charAt(0).toUpperCase() + popupType.slice(1)}</AlertTitle>
          {message}
        </Alert>
      </Snackbar>
    </div>
  );
};

export default Popup;
