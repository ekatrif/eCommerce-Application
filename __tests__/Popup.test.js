import React from 'react';
import renderer from 'react-test-renderer';
import Popup from '../src/components/UI/Popup/Popup';

test('Popup renders correctly', () => {
  const tree = renderer
    .create(<Popup />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
