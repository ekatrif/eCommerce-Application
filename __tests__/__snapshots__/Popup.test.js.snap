// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup renders correctly 1`] = `
<div
  className="popup-active popup"
  onClick={[Function]}
>
  <p>
    Thank you for registering!
  </p>
</div>
`;
